subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformingAsyncArtifactListener.java

```diff
@@ -61,7 +64,7 @@ public boolean includeFileDependencies() {
 
     @Override
     public void fileAvailable(File file) {
-        TransformationSubject initialSubject = TransformationSubject.initial(file);
+        TransformationSubject initialSubject = TransformationSubject.initial(file, configuration);
```

Comment:
This is the entry point where we do not have a `ComponentArtifactIdentifier` available and thus results in having to support a `null` when running the transform and deciding if we can provide the dependencies.

Reply:
IIUC, this is a standalone `File` artifact that has no way to have any dependencies. `TransformationSubject.inital(File)` should not even need the configuration, or anything, because we have no chance to find anything in there.

If we passed around some new type like `DependencyArtitfactLookup` instead of the whole `Configuration` object, we could use a no-op singleton of that here.

Reply:
Done